homeassistant/components/mqtt/mixins.py

```diff
@@ -1347,15 +1347,12 @@ def async_removed_from_device(
     config_entry_id: str,
 ) -> bool:
     """Check if the passed event indicates MQTT was removed from a device."""
-    if event.data["action"] not in ("remove", "update"):
```

Comment:
Due to recent optimizations only `remove` or `update` actions are generated. This code seems unreachable now.

Reply:
I'm not sure how it was possible to get add events in the first place because its subscribing by device_id which couldn't exist if it wasn't added

Reply:
Fyi The tests handle removing entities from a device with multiple entities in several ways. E.g. remove the device from the UI, or remove the device when there are no more associated entities.